src/lib/backend.c

```diff
@@ -5,10 +5,45 @@
 #include "backend_esysdb.h"
 #include "backend_fapi.h"
 
-static bool is_fapi_preview_enabled(void) {
+enum backend {
+    backend_unset = 0,
+    backend_error,
+    backend_esysdb,
+    backend_fapi
+};
+
+static enum backend get_backend(void) {
+
+    /* Using a static to only query env once per runtime. */
+    static enum backend backend = backend_unset;
```

Comment:
So ideally, someone could do:
```C
setenv("TPM2_PKCS11_BACKEND", "1", 1)
C_InitToken()
...
setenv("TPM2_PKCS11_BACKEND", "0", 1)
C_InitToken()
...
```

All in the same process and we want that to work. The query is relatively cheap, getenv is pulling a string mapped in the process memory.

Reply:
ok